tfx/examples/imdb_lstm/imdb_utils_native_keras.py

```diff
@@ -0,0 +1,184 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import List, Text
+
+import absl
+import tensorflow as tf
+from tensorflow import keras
+import tensorflow_transform as tft
+import tensorflow_hub as hub
+
+from tfx.components.trainer.executor import TrainerFnArgs
+
+_TRAIN_BATCH_SIZE = 64 
+_TRAIN_DATA_SIZE = int(50000 * 2 / 3) 
+_EVAL_BATCH_SIZE = 64
+_EVAL_DATA_SIZE = int(50000 / 3)
+_LABEL_KEY = "sentiment"
+DELIMITERS = '.,!?() '
+_MAX_FEATURES = 8000 
+_MAX_LEN = 100 
+
+def _gzip_reader_fn(filenames):
+  """Small utility returning a record reader that can read gzip'ed files."""
+  return tf.data.TFRecordDataset(filenames, compression_type='GZIP'
+          )
+
+def _sentiment_to_int(sentiment):
+    """Converting labels from string to integer"""
+    ints = tf.cast(sentiment, tf.int64)
+    return ints
+
+def _tokenize_review(review):
+    review_sparse = tf.compat.v1.string_split(tf.reshape(review, shape=[-1]), 
+            DELIMITERS)
+    review_indices = tft.compute_and_apply_vocabulary(review_sparse, 
+            default_value=_MAX_FEATURES,
+            top_k=_MAX_FEATURES)
+    dense = tf.sparse.to_dense(review_indices, default_value=_MAX_FEATURES)
+    padding_config = [[0, 0], [0, _MAX_LEN]]
```

Comment:
Lets try using this instead:
padded_inputs = tf.keras.preprocessing.sequence.pad_sequences(raw_inputs,
                                                              padding='post')
Note that we should not actually have to set a max length.

Reply:
worth discussing. I think there'are some issues here.